### ISSUE TITLE
cant run client in windows 11

### ISSUE THREAD

**Issue by kill136** [OpenInterpreter/01#222]
**Describe the bug**
when i run client in windows 11 ,using below commend:
poetry run 01 --client --server-url=172.22.33.166:10001 

i got some log message below:
##########################
(.venv) d:\PycharmProjects\01\software>poetry run 01 --client --server-url=172.22.33.166:10001                                       
Warning: Found deprecated priority 'default' for source 'tsinghua' in pyproject.toml. You can achieve the same effect by changing the priority to 'primary' and putting the source first.
○
Starting...
Hold the spacebar to start recording. Press CTRL-C to exit
[IPKernelApp] WARNING | Parent appears to have exited, shutting down.

######################

**To Reproduce**
1.  i  run  :poetry  run 01 --server   cmd start server, it work well
2. then  i run :poetry run 01 --client --server-url=172.22.33.166:10001 ,it start run then stop 

**Expected behavior**
client can run ok

**Screenshots**
![image](https://github.com/OpenInterpreter/01/assets/22001343/60a01972-0d29-406d-a538-fe26c481dd36)

**Desktop (please complete the following information):**
 - OS:   client : windows 11 ,develop tool  vscode
 - Python Version 3.11

**Additional context**
i make sure the server  is running ok

**Comment by rbrisita** [comment]
I just installed the Windows version but there are bugs like recording and exiting out of the program.

I just ran:

```
poetry run 01
```

Have you tried that?

**Comment by shebarz** [comment]
Yes, I have the same issues on windows. Recording doesn't work and exciting doesn't work. Does anyone know how to fix this?

**Comment by rowaidy** [comment]
> Yes, I have the same issues on windows. Recording doesn't work and exciting doesn't work. Does anyone know how to fix this?

Same here

**Comment by rbrisita** [comment]
There is a PR for exiting out, #231, and the owner has a Windows branch that I have yet to test. Another option would be to run [Docker](https://docs.openinterpreter.com/integrations/docker) if you are familiar with it. Using the link provided could be a starting point to get the 01 to launch.

**Comment by shebarz** [comment]
> There is a PR for exiting out, #231, and the owner has a Windows branch that I have yet to test. Another option would be to run [Docker](https://docs.openinterpreter.com/integrations/docker) if you are familiar with it. Using the link provided could be a starting point to get the 01 to launch.

Thank you! I will try to figure out Docker :) I did not find the windows branch, but I'm also new to github, and that's probably on me

**Comment by rbrisita** [comment]
Welcome to GitHub! Glad to be of help. Here is the branch I was speaking of:

https://github.com/Abdullah-Gohar/01/tree/windows_11

**Comment by rowaidy** [comment]
> Welcome to GitHub! Glad to be of help. Here is the branch I was speaking of:
> 
> https://github.com/Abdullah-Gohar/01/tree/windows_11

Iran it, and got stuck in a loop trying to create a script

**Comment by rbrisita** [comment]
That's called progress! Congrats! And welcome to the point where a lot of us who are not running 01OS under Mac OS X are at.

**Comment by stvn037** [comment]
Any fixes for us windows 10 users? also getting the [https://github.com/OpenInterpreter/01/issues/220](#220) error probably when using server+client terminal. web socket never opens and audio does not upload. including with direct connect to openai.

**Comment by shebarz** [comment]
> Welcome to GitHub! Glad to be of help. Here is the branch I was speaking of:
> 
> https://github.com/Abdullah-Gohar/01/tree/windows_11

Thank you so much! That worked for me :)

**Comment by stvn037** [comment]
> Welcome to GitHub! Glad to be of help. Here is the branch I was speaking of:
> 
> https://github.com/Abdullah-Gohar/01/tree/windows_11

Doesn't work for me on windows 10. After cloning I get "could not find pyproject.toml file". Any tips?

**Comment by rowaidy** [comment]
You need to go into the software directory and then run

**Comment by stvn037** [comment]
I did, and it's not recognized even though it has the original crc and I got past that step in the main branch... curious

**Comment by shebarz** [comment]
> > Welcome to GitHub! Glad to be of help. Here is the branch I was speaking of:
> > https://github.com/Abdullah-Gohar/01/tree/windows_11
> 
> Doesn't work for me on windows 10. After cloning I get "could not find pyproject.toml file". Any tips?

For Windows you have to go in this order and with these commands:

- pip install open-interpreter
- git clone https://github.com/Abdullah-Gohar/01.git   #this is the link to the windows branch
- poetry install
- export OPENAI_API_KEY=sk... 
- poetry run 01


My guess is, you didn't install open interpreter first.
If this doesn work, I think you might not have poetry installed on you machine

**Comment by stvn037** [comment]
well, in the official branch open interpreter was included, so I did not expect to have to install it manually. hope this does the trick. there are several branches there, which one is best for windows 10?

**Comment by Sherwan-Barzange** [comment]
Yeah this is just what worked for me after that error. There is probably a cleaner way, if you're not a noob like me :)

**Comment by stvn037** [comment]
<img width="696" alt="webs-rec" src="https://github.com/OpenInterpreter/01/assets/37275536/e8852faf-80dc-4320-a963-75c3ff5ca0a7">

Not sure what happened before, it doesn't need a sep. interpreter install though.
Now that it's running that build, I still get the same issue anyhow--websocket never opens, recording never uploads.

**Comment by Rob-Milliken** [comment]
Hey, Steven, you are quite correct, open interpreter is installed automagically when running the poetry install. No worries about that. I've been trying to figure out the exact situation you have with specifically *Windows 10*.
What worked for me, and I'd like feedback on if this works for you is to take the logic that detects the OS and force it into *thinking* you have Windows 11. If that works, the powers that be should take note maybe make the logics as such that anyone who has Windows should follow the Windows 11 logic. The file is under source\clients and the file you'll want to make changes is called base_device.py.
Fair warning, once you get it working you'll run into odd code generation loops and only works on macOS modules - but that's another trouble ticket, hopefully you'll see progress and we can get more people with Windows 10 fixing things so it works on this Windows in general (I don't believe it completely does based on what I've seen so far, "out of the box").

![image](https://github.com/OpenInterpreter/01/assets/3958215/46573f42-9aa9-4054-988c-957988f4f440)
![image](https://github.com/OpenInterpreter/01/assets/3958215/a33c66f6-8559-4fc3-82b8-d7903e65603a)

**Comment by stvn037** [comment]
<img width="1311" alt="this" src="https://github.com/OpenInterpreter/01/assets/37275536/7f6ad8e4-1398-4fdb-9547-d5402b23bd60">

Did all that, set the w11 requirement in main project branch---and same issue occurs.
Is this possible related to websockets and/or port forwarding?
really probably need debugging (vscode) or logging to have an idea what's happening.

**Comment by stvn037** [comment]
<img width="626" alt="thisssss" src="https://github.com/OpenInterpreter/01/assets/37275536/3bb7066a-a041-48d3-9249-2a524d98f080">

Not only so, even Windows 12 gets the job done. (: 3
